src/containers/Analytics.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <div>
+    <div class="body-container" :class="{ 'no-data': !hasData() }">
+      <top-bar />
+      <div class="timestamp" v-if="hasData()">Data last retrieved at: {{ analyticsTimestamp }}</div>
+      <pre class="analytics" v-if="hasData()">{{ analyticsData }}</pre>
+      <div class="back_to_home">
+        <a class="back_to_home_link" href="/login">Back to home</a>
+      </div>
+      <custom-footer />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import CustomFooter from '@/components/Footer.vue';
+import TopBar from '@/components/TopBar.vue';
+import { trackUsersCollection } from '@/firebase-frontend-config';
+
+export default defineComponent({
+  components: { CustomFooter, TopBar },
+  mounted() {
+    this.retrieveData();
+  },
+  data() {
+    return {
+      analyticsData: '',
+      analyticsTimestamp: '',
+    };
+  },
+  methods: {
+    retrieveData() {
+      trackUsersCollection.get().then(querySnapshot => {
+        let newestDocData = {};
+        let newestDocDate = new Date(0);
+
+        querySnapshot.forEach(doc => {
+          const isoTimestamp = doc.id;
+          const date = new Date(isoTimestamp);
+          if (date.getTime() > newestDocDate.getTime()) {
+            newestDocDate = date;
+            newestDocData = doc.data();
+          }
+        });
+
+        const output = JSON.stringify(newestDocData, null, 2);
+        this.analyticsData = output;
+        this.analyticsTimestamp = newestDocDate.toLocaleString();
+      });
+    },
```

Comment:
Can we refactor this function, eg. to `global-firestore-data`?

Also, you can potentially make this more efficient: https://stackoverflow.com/a/54178993

Reply:
Refactored and made more efficient, thanks Ben!!